src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoMockStaticToMockito.java

```diff
@@ -42,7 +44,19 @@ public String getDescription() {
 
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
-        return new PowerMockitoToMockitoVisitor();
+        return Preconditions.check(
+                Preconditions.or(
+                    new UsesType<>("org.powermock..*", false),
+                    new UsesType<>("org.mockito..*", false)
+                ),
+                new JavaVisitor<ExecutionContext>(){
+                    @Override
+                    public J visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                        doAfterVisit(new PowerMockitoToMockitoVisitor());
+                        return super.visitClassDeclaration(classDecl, ctx);
+                    }
+                }
```

Comment:
It would probably work.

I chose to go the other way because the current impl of `PowerMockitoToMockitoVisitor` is stateful; it sets a bunch of instance variables in `visitClassDeclaration`. I couldn't quite trace a scenario where this would happen, but it seemed conceivable that "old" values for those instance variables could be used when visiting other syntax elements if the order of visits was not as the original author expected.

So, I used this pattern to ensure that a visit from `PowerMockitoToMockitoVisitor` always begins with a fresh instance, and always starts in the `visitClassDeclaration` method which sets those instance variables.

Ideally, this visitor would probably get a little refactor to avoid using mutable instance variables and therefore avoid this whole question. But in the short term, I could be persuaded either way (my current impl or your proposed change)

Reply:
actually, my current impl is a bit worse than what I just described; it only guarantees that the `PowerMockitoToMockitoVisitor` will visit something which contains a `ClassDeclaration` (the visit will begin from the top as usual), and can stack up multiple `PowerMockitoToMockitoVisitor` visits if the file contains multiple classes. whoops.

taking your suggestion, there's still a slim chance that the recipe will visit a file which meets the mockito/powermockito-usage precondition but also contains top-level method declarations (eg in kotlin or groovy), and then throw that same exception... let me see if I can tweak the code for that

Reply:
pushed a change

Reply:
That helps, thanks! Appreciate you revisiting this one to get it going again.